src/python/TaskWorker/Actions/DagmanResubmitter.py

```diff
@@ -65,7 +65,10 @@ def executeInternal(self, *args, **kwargs): #pylint: disable=unused-argument
         #
         # Processing and tail DAGs will be restarted by these scrips on the
         # schedd after the modifications are made.
-        rootConst = f"(CRAB_DAGType =?= \"BASE\" && CRAB_ReqName =?= {classad.quote(workflow)})"
+        
+        # Only operate on held jobs of this workflow
+        jobConst = f"(CRAB_ReqName =?= {classad.quote(workflow)}) && JobStatus == 5"  # 5 = Held
```

Comment:
Note. As a later refinement we can try to differentiate jobs held by us intentionally from jobs held by condor for whatever other reason. I had not thought about this before, but better to be extra careful on this